core/coreapi/unixfs.go

```diff
@@ -126,6 +131,23 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.File, opts ...options
 		fileAdder.SetMfsRoot(mr)
 	}
 
+	// Get storageMax when node is online.
```

Comment:
Is `Add` the only entry point that will make the repo grow?

Reply:
No. Unfortunately, this _really_ needs to go _much_ lower. We probably need some kind of datastore wrapper that rejects writes when the datastore is full (and calls a user-provided "free" function at N% of full).